src/bot/scrapper/indeed.page.ts

```diff
@@ -0,0 +1,105 @@
+import { JobOffer } from "./types";
+import { ScrapperOptions, Scrapper } from "./scrapper";
+import { Page } from "puppeteer";
+
+export class IndeedScrapper extends Scrapper {
+    
+    private url: string = 'https://www.indeed.com/jobs?q=';
+    private page: Page;
+    private offers: JobOffer[];
+    private jobUrls: string[];
+
+    constructor(
+        options: ScrapperOptions
+    ){
+        super(options);
+        this.jobUrls = [];
+        this.offers = [];
+    }
+
+    private getUrlToScrap(): string{
+        return `${this.url}${this.searchValue.split(' ').join('+')}`;
+    }
+
+    async showOffers(): Promise<JobOffer[]>{
+        return this.offers;
+    }
+
+    async getJobOffer(jobUrl: string): Promise<JobOffer>{
+        await this.navigateToUrl(this.page, jobUrl);
+        await this.page.waitForSelector('#viewJobSSRRoot');
+        let offer = await this.page.$eval('#viewJobSSRRoot', function(element) {
+            function formatSalaryRange(salaryText: string) {
```

Comment:
**nitpick:** This logic is very fragile - I would wrap it into `try...catch` block, and handle errors if they occur. We should inform when something bad happens when scrapping offer